sled-diagnostics/src/logs.rs

```diff
@@ -174,24 +178,32 @@ impl DiagnosticsSnapshot {
 
 impl Drop for DiagnosticsSnapshot {
     fn drop(&mut self) {
-        let _ = Zfs::destroy_snapshot(
-            &self.snapshot.filesystem,
-            &self.snapshot.snap_name,
-        )
-        .inspect(|_| {
-            debug!(
-                self.log,
-                "destroyed sled-diagnostics ZFS snapshot";
-                "snapshot" => %self.snapshot
-            );
-        })
-        .inspect_err(|e| {
-            error!(
-                self.log,
-                "failed to destroy sled-diagnostics ZFS snapshot";
-                "snapshot" => %self.snapshot,
-                "error" => ?e,
-            );
+        let DiagnosticsSnapshot { log, snapshot, .. } = self;
+        tokio::task::spawn({
```

Comment:
To confirm: is it critical that the snapshot **blocks** the drop method? Async drop does not exist in rust.

This API, as implemented, creates the background task but does not await for it to complete.